opennms-webapp/src/main/java/org/opennms/web/tags/DateTimeTag.java

```diff
@@ -0,0 +1,121 @@
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.web.tags;
+
+import static java.time.format.DateTimeFormatter.ISO_LOCAL_DATE;
+import static java.time.temporal.ChronoField.HOUR_OF_DAY;
+import static java.time.temporal.ChronoField.MINUTE_OF_HOUR;
+import static java.time.temporal.ChronoField.SECOND_OF_MINUTE;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.util.Date;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.servlet.jsp.tagext.SimpleTagSupport;
+
+/**
+ * This class replaces the &lt;fmt:formatDate /&gt; tag.
+ * Why do we need a new tag?
+ * => fmt can't be configured via a System Property (without side effects)
+ * => we want to support the new java.time classes
+ *
+ * It will output datetimes as ISO_8601 type style unless otherwise defined in opnnms.properties.
+ * See also:
+ *   https://en.wikipedia.org/wiki/ISO_8601 and
+ *   https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html
+ */
+public class DateTimeTag extends SimpleTagSupport {
+
+    final static String SYSTEM_PROPERTY_DATE_FORMAT = "org.opennms.ui.datettimeformat";
+    private final static Logger log = Logger.getLogger(DateTimeTag.class.getName());
+
+    private final static DateTimeFormatter DEFAULT_FORMATTER = getDefaultFormatter();
+
+
+    private Instant instant;
+    private DateTimeFormatter formatter;
+
+
+    public DateTimeTag(){
+        String format = System.getProperty(SYSTEM_PROPERTY_DATE_FORMAT);
+        if(format == null) {
+            this.formatter = DEFAULT_FORMATTER;
+        } else {
+            try {
+                this.formatter = DateTimeFormatter.ofPattern(format).withZone(ZoneId.systemDefault());
+            } catch (IllegalArgumentException e) {
+                log.log(Level.WARNING,
+                        String.format("Can not use System Property %s=%s as dateformat, will fall back to default." +
+                                " Please see also java.time.format.DateTimeFormatter for the correct syntax",
+                                SYSTEM_PROPERTY_DATE_FORMAT,
+                                format)
+                        , e);
```

Comment:
Is there a particular reason, why we use the `java.util.Logging`instead of the `org.slf4j` logging? 
The usage of String.format in combination with logging looks odd to me.

Reply:
nope, I just saw both in the code so I used what Java offers out of the box. Is org.slf4j  the preferred way? if so I'll change it, np.